packages/jupyter-ai-magics/jupyter_ai_magics/utils.py

```diff
@@ -36,10 +36,9 @@ def get_lm_providers(
             )
             continue
         except Exception as e:
-            log.error(
-                f"Unable to load model provider `{provider_ep.name}`. Printing full exception below."
+            log.warning(
+                f"Unable to load model provider `{provider_ep.name}`", exc_info=e
             )
-            log.exception(e)
```

Comment:
`log.exception(...)` is equivalent to calling `log.error(..., exc_info=exc)`, and "error" is too high a level, so this call needed to change too. The cleanest approach is to just attach the exception instance to the existing `log.warning()` call, as I've done here.

Note, it is [bad practice to use f-strings in logging calls](https://docs.astral.sh/ruff/rules/logging-f-string/#why-is-this-bad), but that should be addressed holistically in a separate PR if desired.

Reply:
I didn't know that about `log.exception()`, thank you!